pds_doi_core/actions/reserve.py

```diff
@@ -103,14 +122,29 @@ def _process_reserve_action_csv(self, target_url):
             if len(dois) == 0:
                 raise InputFormatException("Length of dict_condition_data['dois'] is zero, target_url " + target_url)
 
+            # Add values in _node_long_names as a list of contributors
+            for ii in range(len(dois)):
+                dois[ii].contributors = []
+                for jj in range(len(self._node_long_names)):
+                    dois[ii].contributors.append({'full_name': 'Planetary Data System: ' + self._node_long_names[jj] + ' Node'})
+
             return dois
         except InputFormatException as e:
             logger.error(e)
             exit(1)
+        except OSError as e:
+            msg = f'Error reading file {input_file}'
+            logger.error(msg)
+            raise InputFormatException(msg)
 
     def _parse_input(self, input):
+        # Check for existence first to return the message the 'behave' testing expect.
+        if not os.path.isfile(input):
+            raise InputFormatException(f"Error reading file {input}")
         if input.endswith('.xml'):
-            dois = self._process_reserve_action_pds4(input)
```

Comment:
The 'reserve' action only worked with Excel and CSV output.  If desire we can add it in the future.
In addition, from what I can see from Ron's runs, he was experimenting with XML and JSON and even that, the format as not PDS4 but just XML with the tag same as the column names.  So naming the function _process_reserve_action_pds4 may of been misleading at best.

Reply:
I have added this feature in branch 'doi_for_doc' since the input for reserve were pds4 xml labels. This will be merged after this pull request is merged.